tools/sim/lib/keyboard_ctrl.py

```diff
@@ -14,55 +14,56 @@
 OSPEED = 5
 CC = 6
 
-def getch():
-  fd = sys.stdin.fileno()
-  old_settings = termios.tcgetattr(fd)
+STDIN_FD = sys.stdin.fileno()
+
+def getch() -> str:
+  old_settings = termios.tcgetattr(STDIN_FD)
   try:
     # set
-    mode = termios.tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    #mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN)
+    mode = old_settings.copy()
+    mode[IFLAG] &= ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
+    #mode[OFLAG] &= ~(OPOST)
+    mode[CFLAG] &= ~(CSIZE | PARENB)
+    mode[CFLAG] |= CS8
+    mode[LFLAG] &= ~(ECHO | ICANON | IEXTEN)
     mode[CC][VMIN] = 1
     mode[CC][VTIME] = 0
-    termios.tcsetattr(fd, termios.TCSAFLUSH, mode)
+    termios.tcsetattr(STDIN_FD, termios.TCSAFLUSH, mode)
 
     ch = sys.stdin.read(1)
   finally:
-    termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+    termios.tcsetattr(STDIN_FD, termios.TCSADRAIN, old_settings)
   return ch
 
-def keyboard_poll_thread(q):
+def keyboard_poll_thread(q: 'Queue') -> NoReturn:
   while True:
     c = getch()
     # print("got %s" % c)
     if c == '1':
-      q.put(str("cruise_up"))
-    if c == '2':
-      q.put(str("cruise_down"))
-    if c == '3':
-      q.put(str("cruise_cancel"))
-    if c == 'w':
-      q.put(str("throttle_%f" % 1.0))
-    if c == 'a':
-      q.put(str("steer_%f" % 0.15))
-    if c == 's':
-      q.put(str("brake_%f" % 1.0))
-    if c == 'd':
-      q.put(str("steer_%f" % -0.15))
-    if c == 'q':
+      q.put("cruise_up")
+    elif c == '2':
+      q.put("cruise_down")
+    elif c == '3':
+      q.put("cruise_cancel")
+    elif c == 'w':
+      q.put("throttle_%f" % 1.0)
+    elif c == 'a':
+      q.put("steer_%f" % 0.15)
+    elif c == 's':
+      q.put("brake_%f" % 1.0)
+    elif c == 'd':
+      q.put("steer_%f" % -0.15)
+    elif c == 'q':
       exit(0)
 
-def test(q):
-  while 1:
-    print("hello")
-    time.sleep(1.0)
+def test(q: 'Queue') -> NoReturn:
```

Comment:
why is Queue a string?

Reply:
since it's only imported `if __name__ == '__main__'` i used a [forward reference](https://www.python.org/dev/peps/pep-0484/#forward-references) so we don't have to always import it, the type checkers can figure out that it's a multiprocessing.Queue this way 
![image](https://user-images.githubusercontent.com/17343631/120121441-7bf43800-c171-11eb-801b-3d04e02e625c.png).

you can also use this with TYPE_CHECKING to reduce imports at runtime while still supporting type checkers if it's the only reason you're importing stuff https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING, which i use pretty frequently in my own code.